File: back-end/src/main/java/com/spring/restaurant/dto/PurchaseRequest.java
package com.spring.restaurant.dto;


import java.util.List;
import java.util.Set;

import com.spring.restaurant.model.Address;
import com.spring.restaurant.model.Client;
import com.spring.restaurant.model.Item;
import com.spring.restaurant.model.RequestOrder;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class PurchaseRequest {
	
	private Client client;
	private RequestOrder requestOrder;
	private List<Item> items;
	private Address fromAddress;
	

}

File: frontend/src/app/model/purchase-request.ts
import { Address } from './address';
import { Client } from './client';
import { Item } from './item';
import { RequestOrder } from './request-order';

export class PurchaseRequest {
  client: Client;
  fromAddress: Address;
  requestOrder: RequestOrder;
  items: Item[];

  constructor(
    client: Client,
    fromAddress: Address,
    requestOrder: RequestOrder,
    items: Item[]
  ) {
    this.client = client;
    this.fromAddress = fromAddress;
    this.requestOrder = requestOrder;
    this.items = items;
  }
}
